### ISSUE TITLE
Simplify download code

### ISSUE THREAD

**Issue by huwdp** [golfinq/ActionScript_Event_Builder#1]
The whole documentation can be downloaded here (http://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/standalone.zip) and extracted. Thought this might simplify the code a little :).

How easy would this project be to modify to output options classes (e.g. https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/globalization/CollatorMode.html) and models ones (e.g. https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/globalization/CurrencyParseResult.html)? I'd be happy to help out.

I have listed/gamified the AS3 API support here (https://huwdp.github.io/Flash-Matrix/) and would like to see more things implemented in Ruffle as it takes a long time to add things 1-by-1.

**Comment by golfinq** [comment]
I looked into it and unfortunately, the standalone documentation requires a javascript engine to generate DOM which could then be read by python. I also did a recursive download of the documentation in august, which is now part of this repo [here](https://github.com/golfinq/ActionScript_Event_Builder/blob/main/doc_trunk/doc_pages.7z) I think everything went smoothly with it, but the standalone documentation could absolutely be used to check things - It just would take more time ATM. 

The generating code makes no distinction between different types of classes, which comes from the general template I wrote for a generic ActionScript class which is [here](https://github.com/golfinq/ActionScript_Event_Builder/blob/main/utils/templates.py). That said, more niche cases need to be handled better as I originally wrote it for the `Event` classes, so it's going to take a little effort to generalize the code more. 

As an example, I just added a generated CollatorMode.as to the repo [here](https://github.com/golfinq/ActionScript_Event_Builder/blob/main/as_gen/CollatorMode.as) and you can see that it tried to generate a constructor when it shouldn't of; so it should be a simple fix of adding a {% if... %} to the template. 

Towards your last two points, feel free to poke around! I think I cleaned up the code enough that it shouldn't be too hard to follow whats going on, but definitely make an issue if that isn't the case. That's a really cool website you've made and helps put into perspective with what needs done, thanks for it!

**Comment by golfinq** [comment]
> As an example, I just added a generated CollatorMode.as to the repo [here](https://github.com/golfinq/ActionScript_Event_Builder/blob/main/as_gen/CollatorMode.as) and you can see that it tried to generate a constructor when it shouldn't of; so it should be a simple fix of adding a {% if... %} to the template. 

This is now addressed with the most recent PR which added formatting fixes and the AS3 linter from ruffle

**Comment by golfinq** [comment]
> The whole documentation can be downloaded here (http://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/standalone.zip) and extracted. Thought this might simplify the code a little :).
> 
It did! Now the spider doesn't need to be throttled

> How easy would this project be to modify to output options classes (e.g. https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/globalization/CollatorMode.html) and models ones (e.g. https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/globalization/CurrencyParseResult.html)? I'd be happy to help out.
> 
The project should be able to handle these types of classes, for your example I ran `python .\make_as.py  -i flash.globalization` and got [CollatorMode](https://github.com/golfinq/ActionScript_Event_Builder/blob/0178c2fb74bc42cbe5fed9fa34ac270abd9af18a/as_gen/CollatorMode.as) which looks good and [CurrencyParseResult](https://github.com/golfinq/ActionScript_Event_Builder/blob/0178c2fb74bc42cbe5fed9fa34ac270abd9af18a/as_gen/CurrencyParseResult.as) which needs a small change of removing the `super` call and adding `this._currencyString = symbol` in the constructor

**Comment by huwdp** [comment]
Nice! I had a quick look and the Classes below should be able to be generated and merged into Ruffle quite easily. I'm happy to start writting up tests for this. I'll have a look for any other and see if this cool tool can generate them too :)

- VideoCodec
- H264Level
- VideoStatus
- StageVideoAvailabilityReason
- SoundCodec
- MicrophoneEnhancedMode
- StageVideoAvailability
- AudioOutputChangeReason
- H264Profile
- IMEConversionMode
- SecurityPanel
- Context3DTextureFormat
- Context3DTriangleFace
- Context3DTextureFilter
- Context3DWrapMode
- Context3DMipFilter
- Context3DBufferUsage
- Context3DClearMask
- Context3DProfile
- Context3DRenderMode
- Context3DCompareMode
- Context3DStencilAction
- AudioDecoder
- VideoCodec
- H264Level
- VideoStatus
- StageVideoAvailabilityReason
- SoundCodec
- MicrophoneEnhancedMode
- StageVideoAvailability
- AudioOutputChangeReason
- H264Profile
- IMEConversionMode
- WorkerState
- SystemUpdaterType
- TouchscreenType
- MessageChannelState
- ClipboardTransferMode
- ClipboardFormats
- NetGroupReplicationStrategy
- NetGroupSendResult
- URLRequestMethod
- SharedObjectFlushStatus
- NetGroupReceiveMode
- NetGroupSendMode
- URLLoaderDataFormat
- PrintJobOrientation
- KeyboardType
- KeyLocation
- MultitouchInputMode
- MouseCursor
- CertificateStatus
- BlendMode
- ActionScriptVersion
- InterpolationMethod
- ShaderParameterType
- GradientType
- ShaderPrecision
- SpreadMethod
- PixelSnapping
- StageAlign

**Comment by golfinq** [comment]
> Nice! I had a quick look and the Classes below should be able to be generated and merged into Ruffle quite easily. I'm happy to start writting up tests for this. I'll have a look for any other and see if this cool tool can generate them too :)

This was very helpful! I have now added the raw generated files into the main branch under ~~`options_classes`~~ `output\Options_Draft`

**Comment by golfinq** [comment]
> I'm happy to start writting up tests for this.

I am not sure if tests are needed according to the comment [here](https://github.com/ruffle-rs/ruffle/pull/7597#issuecomment-1211242027) and the following one [here](https://github.com/ruffle-rs/ruffle/pull/7597#issuecomment-1211260599) these are still simple enough to just require manual inspection.

**Comment by huwdp** [comment]
Hi there!

I see that Globalization is missing in here (https://github.com/golfinq/ActionScript_Event_Builder/tree/main/output/Options_Draft/flash).

I also doubled checked for any stuff that I missed out on and found the ones below. They can be added for sure :)
Lets get:
- DisplacementMapFilterMode
- BitmapFilterType
- Context3DBlendFactor
- Context3DVertexBufferFormat
- Context3DProgramType


I think with the changes above, we are ready to merge this into Ruffle :)



------------
I had a look and I believe the stuff below is possible to stub/implement in the future using this tool :)

Many of the things below are stubbed in Lightspark and the stubbs allow games etc to run even though they are not fully implemented.

- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/text/TextRenderer.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/text/StyleSheet.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/text/CSMSettings.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/accessibility/Accessibility.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/H264VideoStreamSettings.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/VideoStreamSettings.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/MicrophoneEnhancedOptions.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/AVNetworkingParams.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/media/AVTagData.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/JPEGLoaderContext.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/IME.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/globalization/NumberParseResult.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/globalization/CurrencyParseResult.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/net/FileFilter.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/net/NetStreamPlayOptions.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/net/URLRequestDefaults.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/net/URLRequestDefaults.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/MouseCursorData.html
- https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Multitouch.html

**Comment by huwdp** [comment]
I'm going to close this :)
